dbt_common/record.py

```diff
@@ -609,12 +616,12 @@ def wrapping_init_subclass(sub_cls):
                         metadata["group"],
                         metadata["index_on_thread_id"],
                         sub_method.__func__
-                        if _is_classmethod(method)
-                        else sub_method,  # unwrap if classmethod
+                        if override_as_classmethod
```

Comment:
Should we go one step further and just make this `getattr(sub_method, "__func__", sub_method)`? Or are there scenarios where `_is_classmethod(method) is False` and `hasattr(sub_method, "__func__") is True`?

Reply:
I set up a test to check for the correct behavior when the parent class has an instance method that is being overriden by a classmethod. However, the fix is slightly risky, and riskier than leaving the limitation in place for now, IMO.

All of which is to say, I'd like to move forward merging this fix, and get to perfect later.